src/PostgREST/AppState.hs

```diff
@@ -281,8 +281,9 @@ internalConnectionWorker appState = work
           -- Fatal error when connecting
           logWithZTime appState reason >> killThread (getMainThreadId appState)
         NotConnected ->
-          -- Unreachable because establishConnection will keep trying to connect
-          return ()
+          -- Unreachable because establishConnection will keep trying to connect, unless disable-recovery is turned on
+          when configDisableRecovery
+              $ logWithZTime appState "Retrying disabled, killing thread..." >> killThread (getMainThreadId appState)
```

Comment:
Mostly to be consistent with the [docs](https://postgrest.org/en/stable/references/connection_pool.html#automatic-recovery). Also I think we don't need the `"..."`, which "convey" that we're going to keep working and in this case we die.

```suggestion
              $ logWithZTime appState "Automatic recovery disabled, exiting." >> killThread (getMainThreadId appState)
```